src/Time/Series.hs

```diff
@@ -162,10 +164,15 @@ instance ( KnownRatName unit
 #endif
                     flooredNewUnit = floorUnit newUnit
                     timeStr = case flooredNewUnit of
-                                   0 -> ""
+                                   Time 0 -> ""
                                    _ -> show flooredNewUnit
-                    nextUnit = newUnit - flooredNewUnit
-                in if nextUnit == 0
+
+#if ( __GLASGOW_HASKELL__ >= 804 )
+                    nextUnit = withRuntimeDivRat @unit @unit $ newUnit -:- flooredNewUnit
+#else
```

Comment:
Damn, I'm waiting for that day when we will drop support for GHC < 8.4.
Or when there appear good way to handle multiple versions.